script/search/2.0/search.inc.php

```diff
@@ -0,0 +1,357 @@
+      $result->pageNumber = $pageNumber;
+
+      if(in_array($platform, array('macos', 'windows', 'linux', 'android', 'ios'))) {
+        $result->platform = $platform;
+      } else if (in_array($platform, ['desktopWeb', 'mobileWeb'])) {
+        $result->platform = 'web';
+      }
+
+      $result->filter = KeyboardSearchResult::FILTER_SEARCH;
+      $result->regionmatch = false;
+
+      $textparts = explode(':', $query);
+
+      for($i = 0; $i < sizeof($textparts) - 1; $i++) {
+        $match = strtolower($textparts[$i]);
+        if(!strcmp($match, 'bcp47')) $result->filter = KeyboardSearchResult::FILTER_LANGUAGE;
+        else if(!strcmp($match, 'id')) $result->filter = KeyboardSearchResult::FILTER_ID;
+        else if(!strcmp($match, 'all')) $result->allmatch = true;
+        else if(!strcmp($match, 'region')) $result->regionmatch = true;
+        else if(!strcmp($match, 'legacy')) $result->filter = KeyboardSearchResult::FILTER_LEGACY;
+      }
+
+      $result->text = array_pop($textparts);
+      $result->searchtext = $result->text;
+
+      return $this->WriteSearchResults($result);
+    }
+
+    private function GetSearchQueries(KeyboardSearchResult $result) {
+      $result->searchtext = strip_tags($result->searchtext);
+
+      switch($result->context) {
+      case KeyboardSearchResult::CONTEXT_KEYBOARD:
+        return $this->LoadKeyboardSearch($result);
+
+      case KeyboardSearchResult::CONTEXT_LANGUAGE:
+        switch($result->filter) {
+        case KeyboardSearchResult::FILTER_ALL:
+          $result->rangetext = "All languages matching '{$result->searchtext}'";
+          $result->languages = $this->LoadLanguageSearch($result->text, 1, $result->allmatch);
+          break;
+        case KeyboardSearchResult::FILTER_ID:
+          $result->rangetext = "Languages with BCP 47 code '{$result->searchtext}'";
+          $result->languages = $this->LoadLanguageSearch($result->text, 1, $result->allmatch);
+          break;
+        case KeyboardSearchResult::FILTER_SEARCH:
+          $result->rangetext = "Languages matching '{$result->searchtext}'";
+          $result->languages = $this->LoadLanguageSearch($result->text, 0, $result->allmatch); //?
+          break;
+        }
+        break;
+
+      case KeyboardSearchResult::CONTEXT_REGION:
+        if($result->regionmatch) {
+          $result->rangetext = "Countries in {$result->region}";
+          $result->countries = $this->LoadRegionSearch($result->text, $result->regionmatch ? 2 : 1);
+        }
+        //else if($result->allmatch && $result->idmatch) $rangetext = "All languages for country with ISO3166-1 code '{$result->searchtext}'";
+        //else if($result->allmatch) $rangetext = "Countries matching '{$result->searchtext}'";
+        else if($result->idmatch) {
+//          $rangetext = "Languages for country with ISO3166-1 code '{$result->searchtext}'";
+//          $langs = $this->LoadLanguageSearch($result->text, 2, $result->allmatch);
```

Comment:
Are these intentionally left in?  Usually we delete dead code, right?

Reply:
I'm planning to do more work on the language and region searches, so haven't cleaned this bit up yet...